Samples/BatchSupport/BatchSupport/main.cpp

```diff
@@ -0,0 +1,90 @@
+﻿#include "pch.h"
+#include "SampleHelper.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::AI::MachineLearning;
+using namespace Windows::Media;
+using namespace Windows::Graphics::Imaging;
+using namespace Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace std;
+
+#define BATCH_SIZE 3
+string modelType = "freeBatchSize";
+string inputType = "TensorFloat";
+
+hstring executionPath =
+    static_cast<hstring>(SampleHelper::GetModulePath().c_str());
+
+bool ParseArgs(int argc, char *argv[]);
+
+int main(int argc, char *argv[]) {
+  init_apartment();
+
+  // did they pass in the args
+  if (ParseArgs(argc, argv) == false) {
+    printf("Usage: %s [fixedBatchSize|freeBatchSize] [TensorFloat|VideoFrame] \n", argv[0]);
+  }
+
+  // load the model
+  hstring modelPath = SampleHelper::GetModelPath(modelType);
+  printf("Loading modelfile '%ws' on the CPU\n", modelPath.c_str());
+  DWORD ticks = GetTickCount();
+  auto model = LearningModel::LoadFromFilePath(modelPath);
+  ticks = GetTickCount() - ticks;
+  printf("model file loaded in %d ticks\n", ticks);
+
+  // now create a session and binding
+  LearningModelDeviceKind deviceKind = LearningModelDeviceKind::Cpu;
+  LearningModelSessionOptions options;
+  if ("freeBatchSize" == modelType) { 
+    // If the model has free dimentional batch, override the free dimension with batch_size
+    options.BatchSizeOverride(static_cast<uint32_t>(BATCH_SIZE));
+  }
+  LearningModelSession session(model, LearningModelDevice(deviceKind), options);
```

Comment:
>session( [](start = 23, length = 8)

In cases of both fix and free batch size, session is created with session options. This isn't what was described in readme.md. It was specified in the readme that session would be created without session options for fixed sized batch dimension models.

Reply:
Yeah, The SessionOptions here for fixed size batch is empty. This is to just make the code cleaner. I will refactor this part and create session in different ways for fixed bacth size and free dimension.

---
In reply to: [379679360](https://github.com/microsoft/Windows-Machine-Learning/pull/295#discussion_r379679360) [](ancestors = 379679360)